src/gtk/textctrl.cpp

```diff
@@ -979,6 +989,83 @@ void wxTextCtrl::GTKSetJustification()
     }
 }
 
+#if wxUSE_SPELLCHECK
+
+wxTextSpellcheckStatus wxTextCtrl::EnableSpellcheck(const bool enable)
+{
+    if (!IsMultiLine())
+        return wxTEXT_SPELLCHECK_NOT_AVAILABLE;
+    
+    GtkTextView *textview = GTK_TEXT_VIEW(m_text);
+    wxASSERT_MSG( textview, wxS("wxTextCtrl is not a GtkTextView" ));
+    
+    wxString lang = wxLocale::GetLanguageCanonicalName(wxLANGUAGE_DEFAULT);
+    
+#if GTK_CHECK_VERSION(3,0,0)
+    GtkSpellChecker *spell = gtk_spell_checker_get_from_text_view(textview);
+
+    if (enable)
+    {
+        if (!spell)
+        {
+            spell = gtk_spell_checker_new();
+            gtk_spell_checker_attach(spell, textview);
+        }
+
+        if (!gtk_spell_checker_set_language(spell, NULL, NULL))
+            return wxTEXT_SPELLCHECK_ERROR;
+    }
+    else
+    {
+        if (spell)
+            gtk_spell_checker_detach(spell);
+    }
+
+#else // GTK_CHECK_VERSION(3,0,0) 
+    // GTK+ 2.x code.
+    GtkSpell *spell = gtkspell_get_from_text_view(textview);
+    GError *err = NULL;
```

Comment:
Please use `wxGtkError` from `include/wx/gtk/private/error.h` to ensure that all errors are freed correctly automatically.